deploy_ticketeer.py

```diff
@@ -0,0 +1,189 @@
+
+    # disable all hosts that are in maintenance
+    disable_tickets_in_maintenance(list(set(hosts_morefs)-set(ready_hosts_morefs)))
+
+    tickets = data.DeployTicket.get({}, conn=conn)
+    # first search for the last FAKE
+    fake_id = search_for_the_last_fake_ticket(tickets)
+
+    # there are only active tickets
+    actual_tickets = list(filter(lambda ticket: int(ticket.id) > int(fake_id), tickets))
+
+    if len(actual_tickets) != vm_per_host * len(hosts):
+        logger.info("ticket imbalance detected...")
+        start_ticket_id = None
+        # create fake ticket that separates old ones and new ones
+        with data.Connection.use('quick') as qc:
+            fake_ticket = data.DeployTicket(
+                created_at=datetime.datetime.now(),
+                host_moref="FAKE",
+                assigned_vm_moref="vm-FAKE",
+                enabled=False
+            )
+            fake_ticket.save(conn=qc)
+            start_ticket_id = fake_ticket.id
+
+        # gets all old ticket ids
+        old_tickets = data.DeployTicket.get({"enabled": "true"}, conn=conn)
+        old_tickets_ids = list(map(lambda ticket: ticket.id,
+                                   filter(lambda ticket: int(ticket.id) < int(start_ticket_id), old_tickets)))
+        # create new tickets, every one in disabled state
+        generate_tickets_in_correct_order(hosts_morefs, vm_per_host)
+
+        for id in old_tickets_ids:
+            with data.Connection.use('quick') as qc:
+                ticket = data.DeployTicket.get_one_for_update({"_id": id}, conn=qc)
+                ticket.enabled = False
+                ticket.save(conn=qc)
+    else:
+        # contiguously enable up to vm_per_host on enabled hosts
+        start_ticket_id = fake_id
+
+        # get all new tickets that are not enabled
+        data_from_db = []
+        with data.Connection.use('conn2') as conn:
+            data_from_db = data.DeployTicket.get({"enabled": 'false'}, conn=conn)
+        new_tickets = list(filter(lambda ticket: int(ticket.id) > int(start_ticket_id), data_from_db))
+
+        # get currently running on each host
+        running = {}
+        for host in ready_hosts:
+            # count running ones
+            with data.Connection.use('conn2') as conn:
+                running[host.mo_ref] = len(data.DeployTicket.get({"host_moref": host.mo_ref, "taken": 1}, conn=conn))
+            # count newly enabled
+            with data.Connection.use('conn2') as conn:
+                running[host.mo_ref] += len(list(
+                    filter(
+                        lambda ticket: int(ticket.id) > int(start_ticket_id),
+                        data.DeployTicket.get({"host_moref": host.mo_ref, 'enabled': 'true'}, conn=conn)
+                    )
+                ))
+        logger.debug(f"running tickets: {running}")
+
+        for ticket in new_tickets:
+            if check_all_running(running, vm_per_host):

```

Comment:
pls consider changing to  `is_ticket_enablement_needed()` (and invert condition)